cuda_bindings/pyproject.toml

```diff
@@ -40,6 +40,16 @@ all = [
     "nvidia-cuda-nvcc-cu12",
     "nvidia-cuda-nvrtc-cu12",
     "nvidia-nvjitlink-cu12>=12.3",
+    "nvidia-cuda-runtime-cu12",
+    "nvidia-cublas-cu12",
+    "nvidia-cufft-cu12",
+    "nvidia-curand-cu12",
+    "nvidia-cusolver-cu12",
+    "nvidia-cusparse-cu12",
+    "nvidia-npp-cu12",
+    "nvidia-nvjpeg-cu12",
+    "nvidia-nvfatbin-cu12",
+    "nvidia-cufile-cu12; sys_platform != 'win32'",
```

Comment:
Just a drive-by comment since the PR is still in draft state. This will be a big interruption to our very slim installation UX. Since we don't provide math libs bindings (it's the job for `nvmath.bindings`), I reckon this is only for testing the path finder, how about moving them to another optional identifier say `test`? Alternatively, we can add a `requirements-test.txt`. (I am hoping setuptools to implement the support for [dependency groups (PEP 735)](https://packaging.python.org/en/latest/specifications/dependency-groups/) which would be yet another good alternative...)

Reply:
> This will be a big interruption to our very slim installation UX.

This isn't meant to be permanent, at all.

What we'll need is separate GHA jobs that install all those wheels just for the purpose of exercising the path_finder.

I'm thinking 2 x 3 jobs:

2: Linux, Windows
3: default CTK installation, conda CTK, nvidia-*-cu12 wheels

We could get away with 2 x 1 (maybe just wheels) for routine CI runs.

2 x 3 if we assign a certain label to the PR.

Reply:
Sounds like a good idea. Do we want this to be tested on a per-PR basis, or nightly (https://github.com/NVIDIA/cuda-python/issues/294) is OK?